packages/next/src/server/render-result.ts

```diff
@@ -105,7 +107,14 @@ export default class RenderResult<
     this.response = response
     this.contentType = contentType
     this.metadata = metadata
-    this.waitUntil = waitUntil
+
+    if (waitUntil) {
+      this.waitUntilQueue.add(() => waitUntil)
+    }
+  }
+
+  public waitUntil(promise: Promise<unknown>) {
```

Comment:
nit: kinda wondering if we should be creating another thing called `waitUntil`... IMHO the `waitUntil` in `pipe-readable` should be called something like `beforeClose` or `beforeEnd`, because it blocks the response from finishing, so it has different semantics from the `waitUntil` function we use for e.g. `after`